cmd/provisioner-localpv/app/config.go

```diff
@@ -338,18 +368,83 @@ func (c *VolumeConfig) IsExt4QuotaEnabled() bool {
 	return enableExt4QuotaBool
 }
 
-//getValue is a utility function to extract the value
+func (c *VolumeConfig) IsPermissionEnabled() bool {
+	permissionEnabled := c.getEnabled(KeyFilePermissions)
+	permissionEnabled = strings.TrimSpace(permissionEnabled)
+
+	permissionEnabledQuotaBool, err := strconv.ParseBool(permissionEnabled)
+	//Default case
+	// this means that we have hit either of the two cases below:
+	//     i. The value was something other than a straightforward
+	//        true or false
+	//    ii. The value was empty
+	if err != nil {
+		return false
+	}
+
+	return permissionEnabledQuotaBool
+}
+
+// GetFsGID fetches the group owner's ID from
+// PVC annotation, if specified
+// NOT YET USED
+func (c *VolumeConfig) GetFsGID() string {
+	if c.IsPermissionEnabled() {
+		configData := c.getData(KeyFilePermissions)
+		if configData != nil {
+			if val, p := configData[KeyFsGID]; p {
+				return strings.TrimSpace(val)
+			}
+		}
+	}
+	return ""
+}
+
+// GetFsGID fetches the user owner's ID from
+// PVC annotation, if specified
+// NOT YET USED
+func (c *VolumeConfig) GetFsUID() string {
+	if c.IsPermissionEnabled() {
+		configData := c.getData(KeyFilePermissions)
+		if configData != nil {
+			if val, p := configData[KeyFsUID]; p {
+				return strings.TrimSpace(val)
+			}
+		}
+	}
+	return ""
+}
+
+// GetFsMode fetches the file mode from PVC
+// or StorageClass annotation, if specified
+func (c *VolumeConfig) GetFsMode() string {
+	if c.IsPermissionEnabled() {
+		configData := c.getData(KeyFilePermissions)
+		if configData != nil {
+			if val, p := configData[KeyFsMode]; p {
+				return strings.TrimSpace(val)
+			}
+		}
+	}
+	//Keep the original default mode
+	return "0777"
```

Comment:
The default value should not be set here. This function should default to an empty value if no valid configuration is found. The default value should be set when this function is used.
```suggestion
	return ""
```